subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -289,6 +295,60 @@ Gradle will treat these directories as possible installations but will not desce
 org.gradle.java.installations.paths=/custom/path/jdk1.8,/shared/jre11
 ----
 
+[NOTE]
+====
+Custom toolchains are not prioritized over <<sec:auto_detection,auto-detected>> ones. If auto-detection is turned on, custom toolchains
+can still be specified but all they will do is to extend the set of detected JRE/JDK locations. Among these locations, one will be picked
+according to the <<sec:precedence,Toolchain Precedence Rules>>.
+====
+
+[[sec:precedence]]
+== Precedence
+
+When there are multiple local JRE/JDK installations matching the toolchain specification of a build, one will be picked according to the
+following rules:
+
+. installation currently running Gradle is preferred over any other
+. JDK locations are preferred over JRE ones
+. certain vendors take precedence over others; their ordering (from the highest priority to lowest):
+.. ADOPTIUM
+.. ADOPTOPENJDK
+.. AMAZON
+.. APPLE
+.. AZUL
+.. BELLSOFT
+.. GRAAL_VM
+.. HEWLETT_PACKARD
+.. IBM
+.. IBM_SEMERU
+.. MICROSOFT
+.. ORACLE
+.. SAP
+.. everything else
+. higher major versions take precedence over lower ones
+. higher minor versions take precedence over lower ones
+. installation paths take precedence according to their lexicographic ordering (last resort criteria for deterministically deciding
+between installations of the same type, from the same vendor and with the same version)
+
+All these rules are applied as multilevel sorting criteria. Let's illustrate with an example. Let's say that the only
+criteria we have added to our toolchain specification is that we are interested in HotSpot implementations and that on our local machine
```

Comment:
We should not do an example that does not include a version since that is not supported.
Also there is not really a criteria for HotSpot, so I would stick with version and vendor for example.

Reply:
Ok. I have done it like this because in the case you describe all matching versions need to have the same major version, so it's a bit limited. But I see your point. Will modify it.